src/js/control-bar/progress-control/seek-bar.js

```diff
@@ -34,7 +33,23 @@ class SeekBar extends Slider {
    */
   constructor(player, options) {
     super(player, options);
-    this.update = Fn.throttle(Fn.bind(this, this.update), 50);
+
+    // when playing, let's ensure we smoothly update the play progress bar
+    // via an interval
+    this.updateInterval = null;
+
+    this.on(player, ['play'], () => {
+      this.clearInterval(this.updateInterval);
+
+      this.updateInterval = this.setInterval(() =>{
+        this.update();
+      }, 15);
+    });
+
+    this.on(player, ['ended', 'pause'], () => {
```

Comment:
This list might not be sufficient. I could potentially see including things like `abort`, `emptied`, `error`, and maybe others. Additionally, I'd be curious to know what happens when including contrib-ads and its event redispatch behavior (such as, would it need to be updated to clear this interval in its own cases).

Reply:
Agreed here. Probably more cases I can cover. I'd like to test these plus others that can be found in `Html5.events` (or something like that - I was grep-happy last night and it came up on my radar).

Reply:
I think the video element is pretty good about firing a pause whenever it's 'not playing', even on ended, but that's worth verifying. Contrib-ads is an interesting question where I have no idea, though this seems like a pretty basic use of play/pause, so if there's a problem with this then contrib-ads probably breaks other things too.

Reply:
Not sure what `contrib-ads` is, but should be an easy search against the codebase. Will update y'all here.

Reply:
Seems like switching to `playing` and adding `waiting` and `stalled` is the way to go. I've updated the PR to include those events.

Reply:
https://html.spec.whatwg.org/multipage/media.html#event-media-playing

Reply:
https://stackoverflow.com/a/14270041/696130

Reply:
@misteroneill @heff do either of y'all know of an easy way to test the redispatch behavior with [contrib-ads](https://github.com/videojs/videojs-contrib-ads/blob/master/src/redispatch.js)? It seems like the only events that get special treatment are `playing`, `ended`, `loadstart`, `loadedata` and `loadedmetadata`. Even though the 2 of that list that we're using is `playing` and `ended`, they seem to be pretty straightforward. It would be nice to do a quick sanity check on my end though.

Reply:
It shouldn't be an issue because the player controls bind the event listeners before contrib-ads's redispatch runs.